### PR TITLE
Increase upper-bound on ghc-prim

### PR DESCRIPTION
**Pull request by shayne-fletcher**
Closes https://github.com/digital-asset/ghc-lib/issues/194.

### CODE REVIEW

File: ghc-lib-gen/src/Ghclibgen.hs [hunk 1]
```diff
@@ -499,7 +499,7 @@ withCommas ms =
 -- | Common build dependencies.
 commonBuildDepends :: [String]
 commonBuildDepends =
-  [ "ghc-prim > 0.2 && < 0.6"
+  [ "ghc-prim > 0.2 && < 0.6.1"
```
Comment (cocreature) [inline ghc-lib-gen/src/Ghclibgen.hs:502]: Why `0.6.1` instead of `0.7`? I would expect ghc-prim to uphold pvp.

Comment (shayne-fletcher) [reply ghc-lib-gen/src/Ghclibgen.hs:502]: I'm changing to 0.7 actually as we speak.

**Comment by shayne-fletcher** [comment]
> lgtm but unless there is a reason for choosing 0.6.1, 0.7 seems to make more sense. This also seems like something that we should test in CI instead of simply bumping the bound.

Yes, going with 0.7. Regarding testing, well, this PR tests it in CI. I do agree though that this aspect of the project needs some love.

**Comment by cocreature** [comment]
What I mean is that we should test that we can build ghc-lib against ghc-prim 0.6 in CI. That doesn’t seem to be part of this PR.

**Comment by shayne-fletcher** [comment]
> What I mean is that we should test that we can build ghc-lib against ghc-prim 0.6 in CI. That doesn’t seem to be part of this PR.

Let's come back to it. For now be assured I am checking these facts locally.

**Comment by shayne-fletcher** [comment]
https://github.com/digital-asset/ghc-lib/issues/197

**Pull request merged**